frontend/src/pages/Assignments/AssignmentRubric/index.tsx

```diff
@@ -0,0 +1,178 @@
+import { useContext, useEffect, useState } from "react";
+import { FaChevronLeft } from "react-icons/fa6";
+import { Link, useLocation } from "react-router-dom";
+
+import "./styles.css";
+import { getRubric, getRubricsInClassroom } from "@/api/rubrics";
+import Button from "@/components/Button";
+import { Table, TableCell, TableRow } from "@/components/Table";
+import { SelectedClassroomContext } from "@/contexts/selectedClassroom";
+import { setAssignmentRubric } from "@/api/assignments";
+
+
+const AssignmentRubric: React.FC = () => {
+  const location = useLocation();
+  const { selectedClassroom } = useContext(SelectedClassroomContext)
+
+
+  const [assignment, setAssignmentData] = useState<IAssignmentOutline>()
+  const [rubricData, setRubricData] = useState<IFullRubric>()
+  const [rubrics, setRubrics] = useState<IFullRubric[]>([])
+
+  const [importing, setImporting] = useState(false)
+
+
+  const choseExisting = async (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedId = parseInt(event.target.value, 10);
+    if (selectedId) {
+      const selectedRubric = rubrics.find((rubric) => rubric.rubric.id === selectedId);
+      if (selectedRubric) {
+        setRubricData(selectedRubric)
+        if (selectedClassroom && assignment) {
+          setAssignmentRubric(
+            selectedRubric.rubric.id!,
+            selectedClassroom.id,
+            assignment.id)
+    
+        }
+      }
+    }
+  };
+
+
+  const notImporting = () => {
+    setImporting(false)
+  }
+
+  const allowImporting = () => {
+    setImporting(true)
+  }
+
+
+  useEffect(() => {
+    if (location.state) {
+      setAssignmentData(location.state.assignment)
+      const aData = location.state.assignment
+      if (aData && aData.rubric_id) {
+        (async () => {
+          try {
+            const rubric = await getRubric(aData.rubric_id)
+            if (rubric !== null) {
+              setRubricData(rubric)
+            }
+          } catch (_) {
+            //do nothing
```

Comment:
these do nothings are placeholders for better error handling. Maybe we could add an error state to this component/page?